### ISSUE TITLE
Incorrect Link href with basepath

### ISSUE THREAD

**Issue by giuem** [molefrog/wouter#76]
Hello, I'm trying to implement basepath into my application. I followed the README's [instruction](https://github.com/molefrog/wouter#customizing-the-location-hook). Everything works well, except **`<Link />`** component.

For example,

``` jsx
<Link href="/path">link</Link>
```
It's fine that clicking on it will navigate to `:basepath/path`, but the href prop is still `/path`.

Now I had to use `<Link />` in this way,

``` jsx
<Link href="/path">
  <span><a href={`${basepath}/path`}>link</a></span>
</Link>
```

Any idea on it?

**Comment by molefrog** [comment]
@giuem Oh yeah, that's a bummer.

You're right since wouter is based on simple `useLocation` subscription the Link doesn't know anything about the path substitution. It should work well in the app, however I agree, if you need to have semantic links for the SSR that would mess things up. As a temp hack I can offer you to either always use absolute urls everywhere, or write your own `Link` implementation that takes care of all the transformations needed.

**Comment by molefrog** [comment]
Another idea is to introduce a new prop on `Link` component say `originalHref` (any other suggestions on the name?) that will control the href. So you then could write a helper:

```js
const linkPropsTo = (path) => ({
  href: path,
  originalHref: basePath + path
})

<Link {...linkPropsTo("/users")} />
```

Just an idea.
```

**Comment by giuem** [comment]
`originalHref` sounds great for me! Or maybe we can introduce `createLink` HOC:

``` js
function createLink(basepath = '') {
  function Link(props) {
    // ...
    const href = props.href || props.to;
    const originalHref = basepath + href;
    // ...
  }
  return Link;
}

const Link = createLink(basepath);

<Link href="/users" />
```

**Comment by omgovich** [comment]
@molefrog Should we close the issue?
As far as I understand the problem was solved here: #94

**Comment by molefrog** [comment]
@omgovich Not yet, there are a few things left:
  - Add types that describe the `basepath` option
  - Document the feature in README (that's on me)

I also haven't decided yet which name to use. Been thinking that `base` is smaller and will help us to save a couple of extra bytes, but not sure if it's clear enough:
```js
<Router base="/app">...</Router>
```

Does this look ok @omgovich @giuem @cbbfcd ?

**Comment by omgovich** [comment]
@molefrog `base` is used by Vue Router and NuxtJS: https://router.vuejs.org/ru/api/#mode
So I think this is OK

I can rename `basepath` to `base` everywhere and send a PR.

**Comment by cbbfcd** [comment]
i agree!!!! good idea!! 😄

**Comment by molefrog** [comment]
@omgovich Ok, thanks. @cbbfcd has just sent a PR with types, we're almost there with merging it. Let's rename the parameter once types land in master.

**Comment by molefrog** [comment]
@omgovich Types have been landed in `master`. Can you rename the param?

**Comment by omgovich** [comment]
@molefrog Sure)

**Comment by molefrog** [comment]
Updated the README: https://github.com/molefrog/wouter#i-deploy-my-app-to-the-subfolder-can-i-specify-a-base-path

**Comment by molefrog** [comment]
The basepath support has just landed in the latest `2.4.0` version! Great work, guys, thanks everyone for the help!
